### PR TITLE
enhancement #114: added new EventType enum

### LINKED ISSUE #114: [ENHANCE] Add new EventType enum
For now, we use DisciplineTypeEnum from database in events, when performing GET requests on shedule events. When we have event that is not connected, we send type = null. It's not a good approach, considering other values are from enum. So, we need to create new enum in code (not in database) that has the same values as DisciplineTypeEnum + value OTHER. Also, you need to change the code and send OTHER value instead of null when there is no connected discipline type. Finally, set this enum in swagger documentation

### PR DESCRIPTION
**Pull request by kol-oss**
- added EventTypeEnum enum
- replaced DisciplineTypeEnum with EventTypeEnum
- replaced sending null with EventTypeEnum.OTHER
- updated Swagger documentation

closes #114

### CODE REVIEW

**Review by ArtemLyah — DISMISSED** [review]

File: fictadvisor-api/src/v2/api/controllers/DisciplineController.ts [hunk 1]
```diff
@@ -116,16 +117,16 @@ export class DisciplineController {
...
   })
   @Access(PERMISSION.GROUPS_$GROUPID_DISCIPLINES_TEACHERS_GET, GroupByDisciplineGuard)
   @Get('/:disciplineId/teachers')
   async getAllByDiscipline (
     @Param('disciplineId', DisciplineByIdPipe) disciplineId: string,
-    @Query('disciplineType') disciplineType: DisciplineTypeEnum,
+    @Query('disciplineType') disciplineType: EventTypeEnum,
   ) {
-    const teachers = await this.disciplineService.getTeachers(disciplineId, disciplineType);
+    const teachers = await this.disciplineService.getTeachers(disciplineId, disciplineType as DisciplineTypeEnum ?? null);
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/controllers/DisciplineController.ts:129]: You should only use EventTypeEnum with schedule logic

File: fictadvisor-api/src/v2/api/dtos/AttachLessonDTO.ts [hunk 2]
```diff
@@ -14,11 +14,11 @@ export class AttachLessonDTO {
...
 
   @ApiProperty({
-    enum: DisciplineTypeEnum,
+    enum: EventTypeEnum,
   })
-  @IsEnum(DisciplineTypeEnum, validationOptionsMsg('Discipline type must be an enum'))
+  @IsEnum(EventTypeEnum, validationOptionsMsg('Discipline type must be an enum'))
   @IsNotEmpty()
-    disciplineType: DisciplineTypeEnum;
+    disciplineType: EventTypeEnum;
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/dtos/AttachLessonDTO.ts:21]: eventType

File: fictadvisor-api/src/v2/api/dtos/CreateEventDTO.ts [hunk 3]
```diff
@@ -21,11 +22,11 @@ export class CreateEventDTO {
...
 
   @ApiPropertyOptional({
-    enum: DisciplineTypeEnum,
+    enum: EventTypeEnum,
   })
   @IsOptional()
-  @IsEnum(DisciplineTypeEnum, validationOptionsMsg('Discipline type must be an enum'))
-    disciplineType?: DisciplineTypeEnum;
+  @IsEnum(EventTypeEnum, validationOptionsMsg('Discipline type must be an enum'))
+    disciplineType?: EventTypeEnum;
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/dtos/CreateEventDTO.ts:29]: eventType

File: fictadvisor-api/src/v2/api/dtos/UpdateEventDTO.ts [hunk 4]
```diff
@@ -35,11 +36,11 @@ export class UpdateEventDTO {
...
 
   @ApiPropertyOptional({
-    enum: DisciplineTypeEnum,
+    enum: EventTypeEnum,
   })
   @IsOptional()
-  @IsEnum(DisciplineTypeEnum, validationOptionsMsg('Discipline type must be an enum'))
-    disciplineType?: DisciplineTypeEnum;
+  @IsEnum(EventTypeEnum, validationOptionsMsg('Discipline type must be an enum'))
+    disciplineType?: EventTypeEnum;
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/dtos/UpdateEventDTO.ts:43]: eventType

File: fictadvisor-api/src/v2/api/responses/DisciplineTypeResponse.ts [hunk 5]
```diff
@@ -11,10 +11,10 @@ export class DisciplineTypeResponse {
   })
     disciplineId: string;
   @ApiProperty({
-    enum: DisciplineTypeEnum,
+    enum: EventTypeEnum,
     description: 'Discipline type',
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/responses/DisciplineTypeResponse.ts:15]: Fix Swagger doc

File: fictadvisor-api/src/v2/api/responses/MainEventInfoResponse.ts [hunk 6]
```diff
@@ -25,10 +26,10 @@ export class SimpleEventInfoResponse {
 
 export class MainEventInfoResponse extends SimpleEventInfoResponse {
   @ApiProperty({
-    type: DisciplineTypeResponse,
+    type: DisciplineTypeResponse || EventTypeEnum,
     description: 'Type of event',
   })
-    disciplineType: DisciplineTypeResponse;
+    disciplineType: DisciplineTypeResponse | EventTypeEnum;
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/responses/MainEventInfoResponse.ts:32]: only EventType

File: fictadvisor-api/src/v2/api/responses/EventResponse.ts [hunk 7]
```diff
@@ -26,9 +27,9 @@ export class EventResponse {
     disciplineId: string;
 
   @ApiProperty({
-    enum: DisciplineTypeEnum,
+    enum: EventTypeEnum,
   })
-    disciplineType: DisciplineTypeEnum;
+    disciplineType: EventTypeEnum;
```
Comment (fokaaas) [inline fictadvisor-api/src/v2/api/responses/EventResponse.ts:32]: eventType

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by fokaaas — DISMISSED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**